sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/bitwiseAggregates.scala

```diff
@@ -97,3 +97,42 @@ case class BitOrAgg(child: Expression) extends DeclarativeAggregate with Expects
 
   override lazy val evaluateExpression: AttributeReference = bitOr
 }
+
+@ExpressionDescription(
+  usage = "_FUNC_(expr) - Returns the bitwise XOR of all non-null input values, or null if none.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(col) FROM VALUES (3), (5) AS tab(col);
+       6
+  """,
+  since = "3.0.0")
+case class BitXorAgg(child: Expression) extends DeclarativeAggregate with ExpectsInputTypes {
```

Comment:
`BitAndAgg`, `BitOrAgg`, and `BitXorAgg` has the similar logic, so can we share it by using a new trait (e.g., BitwiseOpLike)?

Reply:
sounds good, will update

Reply:
I think you can make it much simpler like this;
```
abstract class BitAggregate extends DeclarativeAggregate with ExpectsInputTypes {

  ...
  override lazy val updateExpressions: Seq[Expression] =
    If(IsNull(bitAgg),
      child,
      If(IsNull(child), bitAgg, bitOp(bitAgg, child))) :: Nil

  override lazy val mergeExpressions: Seq[Expression] =
    If(IsNull(bitAgg.left),
      bitAgg.right,
      If(IsNull(bitAgg.right), bitAgg.left, bitOp(bitAgg.left, bitAgg.right))) :: Nil
}

case class BitAndAgg(child: Expression) extends BitAggregate {

  override def nodeName: String = "bit_and"
  override def bitOp(left: Expression, right: Expression): BinaryArithmetic =
    BitwiseAnd(left, right)
}
```

Reply:
yea, much cooler, thanks